Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -508,6 +509,60 @@ protected virtual GrpcChannelOptions GetChannelOptions()
         /// </summary>
         public abstract TClient Build();
 
+        /// <summary>
+        /// Populates properties based on those set via dependency injection.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Credentials are only requested from dependency injection if they are not already set
+        /// via any of <see cref="ChannelCredentials"/>, <see cref="CredentialsPath"/>,
+        /// <see cref="JsonCredentials"/>, <see cref="Credential"/>, <see cref="GoogleCredential"/>
+        /// or <see cref="TokenAccessMethod"/>.
+        /// </para>
+        /// <para>
+        /// If credentials are requested, they are tried in the following order:
+        /// </para>
+        /// <list type="bullet">
+        /// <item>ChannelCredentials</item>
+        /// <item>ICredential</item>
+        /// <item>GoogleCredential</item>
+        /// </list>
+        /// </remarks>
+        /// <param name="provider">The service provider to request dependencies from.</param>
+        protected virtual void Configure(IServiceProvider provider)
```

Comment:
We need to remember to handwrite the extension methods for the libraries that have builder wrappers, I think PubSub is one of them?